### PR TITLE
Prevent atomizer write file if CSS doesn't change

### PR DESCRIPTION
**Pull request by roderickhsiao**
@tom76kimo 

The issue we encountered is that we are watching atomic CSS and restart the server if needed to deliver atomic css server side. a lot of time there is no new atomic CSS generated but as the plugin always trying to write the file, which trigger the watch process and is very expensive on development.

The idea is to have a cached string so we only try to write file when the CSS file changes (can also avoid file write to speed up the build process)

Also update unit test which never really run correctly. 

c.c. @tinder-achan

### CODE REVIEW

File: tests/atomicLoader.js [hunk 1]
```diff
@@ -1,21 +1,16 @@
...
+describe('atomic loader', () => {
+  it('can generate correct css', (done) => {
+    this.async = () => (err, source) => {
+      var cssReg = new RegExp(/\.Bgc\\\(yellow\\\)/);
+      const cssFile = fs.readFileSync('./build/css/atomic.css');
+      expect(cssReg.test(cssFile)).to.equal(true);
+      done();
+    };
+    atomicLoader.call(this, '<div class="Bgc(yellow)"></div>');
+  });
```
Comment (tom76kimo) [inline tests/atomicLoader.js]: True, thanks for the updates.

**Comment by tom76kimo** [comment]
I like the idea, thanks

File: lib/atomicLoader.js [hunk 2]
```diff
@@ -93,8 +96,16 @@ const parseAndGenerateFile = function(
 
         pipeline.process(cssString).then(result => {
             const { css = '' } = result;
+
+            if (css === cachedResponse) {
```
Comment (tom76kimo) [inline lib/atomicLoader.js]: Wondering if `pipeline.process` aways generate new `css` or not. If `css` aways have a new memory address, `===` should always return `false`?

**Comment by tom76kimo** [comment]
Has publish this PR with version `0.5.0`

File: lib/atomicLoader.js (hunk 2 above)
Comment (roderickhsiao) [reply lib/atomicLoader.js]: I think the `css` is just string not stream, so the comparison will just be primitive value comparison

**Comment by roderickhsiao** [comment]
Thanks Tom

File: lib/atomicLoader.js (hunk 2 above)
Comment (roderickhsiao) [reply lib/atomicLoader.js]: I tried locally and do see the comparison running correctly :) I am debating should we reset the cached css, but as we are not comparing the length, so should be fine

Comment (tom76kimo) [reply lib/atomicLoader.js]: Great! Thanks!

Comment (tom76kimo) [reply lib/atomicLoader.js]: Great! Thanks!

**Pull request merged**